File: src/tools/listModels.ts
import { listModels as listModelsCatalog } from '../models.js';
import type { ListModelsInput, ListModelsResponse } from '../types.js';

export async function listModels(input: ListModelsInput = {}): Promise<ListModelsResponse> {
  const taskType = input.task_type ?? 'all';
  return {
    models: listModelsCatalog(taskType).map((model) => ({
      model_id: model.modelId,
      label: model.label,
      task_type: model.taskType,
      cost_tier: model.costTier,
      recommended: model.recommended,
    })),
  };
}

File: handlers/list_models.py
from __future__ import annotations

from core.models_catalog import list_models
from models.inputs import ListModelsInput
from models.outputs import ListModelsResponse, ModelInfo


def handle_list_models(input: ListModelsInput) -> ListModelsResponse:
    return ListModelsResponse(
        models=[
            ModelInfo(
                model_id=model.model_id,
                label=model.label,
                task_type=model.task_type,
                cost_tier=model.cost_tier,
                recommended=model.recommended,
            )
            for model in list_models(input.task_type)
        ]
    )
